crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -886,6 +887,27 @@ declare_lint! {
     }
 }
 
+declare_lint! {
+    /// ## What it does
+    /// Checks for the creation of invalid `NewType`s
+    ///
+    /// ## Why is this bad?
+    /// There are several requirements that you must follow when creating a `NewType`.
```

Comment:
I think if I were a user I would immediately have the question here "...but _why_ must I follow these requirements?" 😆 I think we should either explain a bit more clearly here why the rules around `NewType` are the way they are, or link to some docs elsewhere that give that explanation